src/main/kotlin/com/ampnet/reportservice/service/impl/XlsxServiceImpl.kt

```diff
@@ -0,0 +1,144 @@
+        }
+    }
+
+    private fun getUsers(coop: String, type: XlsxType): List<UserResponse> =
+        when (type) {
+            XlsxType.REGISTERED -> userService.getAllUsers(coop)
+            XlsxType.VERIFIED -> mapUserResponse(userService.getAllActiveUsers(coop).usersList)
+            XlsxType.WALLET -> {
+                // users with initialized wallet
+                val activeUsers = userService.getAllActiveUsers(coop)
+                val walletOwners = walletService
+                    .getWalletsByOwner(activeUsers.usersList.map { UUID.fromString(it.uuid) })
+                    .map { it.owner }
+                val usersWithWallet = activeUsers.usersList.filter { walletOwners.contains(it.uuid) }
+                mapUserResponse(usersWithWallet)
+            }
+            XlsxType.DEPOSIT -> {
+                // users with approved deposit
+                val depositOwners = walletService.getOwnersWithApprovedDeposit(coop)
+                val usersWithDeposit = userService.getAllActiveUsers(coop).usersList
+                    .filter { depositOwners.contains(it.uuid) }
+                mapUserResponse(usersWithDeposit)
+            }
+            XlsxType.INVESTMENT -> {
+                // users with at least one investment
+                val activeUsers = userService.getAllActiveUsers(coop).usersList
+                val walletsWithInvestment = blockchainService.getUserWalletsWithInvestment(coop).map { it.wallet }
+                val walletOwners = walletService
+                    .getWalletsByOwner(activeUsers.map { UUID.fromString(it.uuid) })
+                    .filter { walletsWithInvestment.contains(it.activationData) }
+                    .map { it.owner }
+                val usersWithInvestment = activeUsers.filter { walletOwners.contains(it.uuid) }
+                mapUserResponse(usersWithInvestment)
+            }
+        }
+
+    private fun writeHeaderLine(type: XlsxType) {
+        sheet = workbook.createSheet("Users-$type")
+        val font = workbook.createFont().apply {
+            bold = true
+            fontHeight = headerFontSize
+        }
+        val style = workbook.createCellStyle().apply { setFont(font) }
+        val row = sheet.createRow(0)
+        var columnCount = 0
+        createCell(row, columnCount++, "User UUID", style)
+        createCell(row, columnCount++, "E-mail", style)
+        createCell(row, columnCount++, "First Name", style)
+        createCell(row, columnCount++, "Last Name", style)
+        createCell(row, columnCount++, "Authentication Method", style)
+        createCell(row, columnCount, "Registration Date", style)
+    }
+
+    private fun writeDataLines(users: List<UserResponse>) {
+        val font = workbook.createFont().apply { fontHeight = dataFontSize }
+        val style = workbook.createCellStyle().apply { setFont(font) }
+        var rowCount = 1
```

Comment:
Columns start from 0 index, but this starts from 1 - is this correct or a mistake?

Reply:
The first column is for table header values.